google-cloud-examples/src/main/java/com/google/cloud/examples/pubsub/snippets/PublisherClientSnippets.java

```diff
@@ -0,0 +1,164 @@
+      // [START listTopicSubscriptions]
+      TopicName topicName = TopicName.create(projectId, topicId);
+      ListTopicSubscriptionsRequest request =
+          ListTopicSubscriptionsRequest.newBuilder()
+              .setTopicWithTopicName(topicName)
+              .build();
+      ListTopicSubscriptionsPagedResponse response =
+          publisherClient.listTopicSubscriptions(request);
+      Iterable<String> subscriptionNames = response.iterateAllElements();
+      for (String subscriptionName : subscriptionNames) {
+        // do something with the subscription name
+      }
+      // [END listTopicSubscriptions]
+      return response;
+    }
+  }
+
+  /** Example of deleting a topic. */
+  public TopicName deleteTopic(String topicId) throws Exception {
+    try (PublisherClient publisherClient = PublisherClient.create()) {
+      // [START deleteTopic]
+      TopicName topicName = TopicName.create(projectId, topicId);
+      publisherClient.deleteTopic(topicName);
+      // [END deleteTopic]
+      return topicName;
+    }
+  }
+
+  /** Example of getting a topic policy. */
+  public Policy getTopicPolicy(String topicId) throws Exception {
+    try (PublisherClient publisherClient = PublisherClient.create()) {
+      // [START getTopicPolicy]
+      TopicName topicName = TopicName.create(projectId, topicId);
+      Policy policy = publisherClient.getIamPolicy(topicName.toString());
+      if (policy == null) {
+        // topic iam policy was not found
+      }
+      // [END getTopicPolicy]
+      return policy;
+    }
+  }
+
+  /** Example of replacing a topic policy. */
+  public Policy replaceTopicPolicy(String topicId) throws Exception {
+    try (PublisherClient publisherClient = PublisherClient.create()) {
+      // [START replaceTopicPolicy]
+      String topicName = TopicName.create(projectId, topicId).toString();
+      Policy policy = publisherClient.getIamPolicy(topicName);
+      // add role -> members binding
+      Binding binding =
+          Binding.newBuilder()
+              .setRole(Role.viewer().toString())
+              .addMembers(Identity.allAuthenticatedUsers().toString())
+              .build();
+      // create updated policy
+      Policy updatedPolicy = Policy.newBuilder(policy).addBindings(binding).build();
+      updatedPolicy = publisherClient.setIamPolicy(topicName, updatedPolicy);
+      // [END replaceTopicPolicy]
+      return updatedPolicy;
+    }
+  }
+
+  /** Example of testing whether the caller has the provided permissions on a topic.
+   * Only viewer, editor or admin/owner can view results of pubsub.topics.get  */
+  public TestIamPermissionsResponse testTopicPermissions(String topicId) throws Exception {
+    try (PublisherClient publisherClient = PublisherClient.create()) {
+      // [START testTopicPermissions]
+      List<String> permissions = new LinkedList<>();
+      permissions.add("pubsub.topics.get");
+      TopicName topicName = TopicName.create(projectId, topicId);
```

Comment:
I updated to have projectId get picked up from `ServiceOptions.getDefaultProjectId()`. The snippets will still show `projectId` but the constructor in the snippet shows where  we pick it up from.